contracts/libraries/NFTSVG.sol

```diff
@@ -369,4 +376,28 @@ library NFTSVG {
             return ('24', '27');
         }
     }
+
+    function generateSVGRareSparkle(uint256 tokenId, address poolAddress) private pure returns (string memory svg) {
+        if (isRare(tokenId, poolAddress)) {
+            svg = string(
+                abi.encodePacked(
+                    '<g style="transform:translate(226px, 392px)"><rect width="36px" height="36px" rx="8px" ry="8px" fill="none" stroke="rgba(255,255,255,0.2)" />',
+                    '<path style="transform:translate(6px,6px)" d="M12 0L12.6522 9.56587L18 1.6077L13.7819 10.2181L22.3923 6L14.4341 ',
+                    '11.3478L24 12L14.4341 12.6522L22.3923 18L13.7819 13.7819L18 22.3923L12.6522 14.4341L12 24L11.3478 14.4341L6 22.39',
+                    '23L10.2181 13.7819L1.6077 18L9.56587 12.6522L0 12L9.56587 11.3478L1.6077 6L10.2181 10.2181L6 1.6077L11.3478 9.56587L12 0Z" fill="white"/></g>'
+                )
+            );
+        } else {
+            svg = '';
+        }
+    }
+
+    function isRare(uint256 tokenId, address poolAddress) internal pure returns (bool) {
+        return getHash(tokenId, poolAddress) < type(uint256).max / (1 + BitMath.mostSignificantBit(tokenId) * 2);
+    }
+
+    function getHash(uint256 tokenId, address poolAddress) internal pure returns (uint256) {
+        bytes32 h = sha256(abi.encodePacked(tokenId, poolAddress));
```

Comment:
is this different from keccak256? any reason to use one or the other?

Reply:
nope, updated though. saved gas

Reply:
this func is tiny and used in one place, does it make sense to inline? also don't see the hash function update is pushed

Reply:
you must be looking at a commit?  (these comments are listed as outdated) Will do, though